docs/cookbook/images-gridfs.rst

```diff
@@ -0,0 +1,643 @@
+How to store images in MongoDB GridFS?
+======================================
+
+This guide will show you how to store product images in MongoDB GridFS using the
+`DoctrineMongoDBBundle <https://symfony.com/doc/current/bundles/DoctrineMongoDBBundle/index.html>`_.
+We're assuming that you already enabled and configured the bundle accordingly.
+
+Prerequisite: data structure
+----------------------------
+
+First of all a document class to store an image is required. This example assumes the ``DoctrineMongoDBBundle`` is configured
+using the annotation driver.
+
+To make GridFS files easily reusable, we introduce a mapped superclass which will store the basic file information.
+
+.. code-block:: php
+
+    namespace AppBundle\Document;
+
+    use Doctrine\MongoDB\GridFSFile;
+    use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+
+    /**
+     * @ODM\MappedSuperclass
+     */
+    abstract class File
+    {
+        /**
+         * @ODM\Id
+         *
+         * @var string
+         */
+        private $id;
```

Comment:
My intention is to avoid accessing the members directly as they are immutable and should only be accessed by it's getter. But this is mainly a convention used in the project this examples are taken of. If it's not compatible with Sylius I can change it - it's an example implementation only ;)